### ISSUE TITLE
GTFS Guidelines: updates to Trip Updates and Vehicle Positions feeds

### ISSUE THREAD

**Issue by evansiroky** [cal-itp/data-infra#567]
## Question

Have the Trip Updates and Vehicle Positions feeds been updated every 20 seconds for the past week?

## Metrics

* Updates to Trip Updates feed
* Updates to Vehicle Positions feed

## Data sources

* Trip Updates feed
* Vehicle Positions feed

## (Data Servicess Team to Copy and Fill Out Below)

The QuVR MD template below will be filled out by a member of the data services team.
This allows us to describe the request, in a way that is easy to hand-off for analysis.
After the research phase, we will sync with the asker to figure out if the metric and dashboard pieces are needed.

Before starting research:

* **Question**: Question written as a single sentence.
* **View**: E.g. `views.gtfs_schedule_fact_daily_feed_files`.
* **Research**:
  - How should the results be presented?
  - When are they needed by?

After reviewing research with the asker:

* **Metric**: what specific calculations are needed?
* **Dashboard**: where should we put the result?

**Comment by machow** [comment]
From pairing with @evansiroky, there are a number of challenges to measuring this guideline:

* We sample feeds every 20 seconds, which is insufficient to detect whether something is updating every 20 seconds (we would need to sample every 10 seconds or shorter). So we will need to relax our criteria.
  - e.g. detect "X updates every Y minutes", where it is something less than 3 a minute. 
  - could use a rolling window, etc..
* Research is needed to identify the exact method for detecting an updated feed.
  - we have a big list of all feed files we download, along with a time downloaded and hash
  - feed files have a header timestamp, and often timestamps in specific entries
  - we need to explore the relationship between header and time downloaded (e.g. do some people send back old info with a new header timestamp? does header timestamp ever do weird things like say it's in the future?)

**Comment by evansiroky** [comment]
Proposed way to answer question: for each attempt (hopefully every 20 seconds, but no big deal if 21-40 seconds) was the attempt successful (successful as in feed downloaded and the root `FeedHeader#timestamp` is greater than the last observed `FeedHeader#timestamp`). Aggregate this to a daily (maybe quarterly/monthly/etc) metric of percent of percent of successful attempts.

**Comment by evansiroky** [comment]
Do this for Alerts to for the heck of it.

**Comment by evansiroky** [comment]
Proposed daily fact table:

- calitp_id
- url_number
- date
- rt feed type (alerts, trip updates or vehicle positions)
- number of attempts
- number of successes

**Comment by evansiroky** [comment]
Closing in favor of #2556.
